pipe_test.go

```diff
@@ -37,6 +38,26 @@ func TestDialListenerTimesOut(t *testing.T) {
 	}
 }
 
+func TestDialListenerGetsCancelled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	l, err := ListenPipe(testPipeName, nil)
```

Comment:
Thank you for the test!

Can you also add a timeout test for the `DialPipeContext` call? Similar to `TestDialListenerTimesOut`